crates/apollo-compiler/src/validation/validation_db.rs

```diff
@@ -135,50 +156,82 @@ pub fn check_union_type_definition(
 }
 
 pub fn check_enum_type_definition(
-    _db: &dyn ValidationDatabase,
-    _enum_type: Arc<hir::EnumTypeDefinition>,
+    db: &dyn ValidationDatabase,
+    enum_def: Arc<hir::EnumTypeDefinition>,
 ) -> Vec<ApolloDiagnostic> {
-    // TODO: validate extensions
-    vec![]
+    let mut diagnostics = Vec::new();
+
+    for val in enum_def.enum_values_definition() {
+        diagnostics.extend(db.check_enum_value(val.clone()))
+    }
+
+    diagnostics
 }
 
-pub fn check_input_object_type_definition(
-    _db: &dyn ValidationDatabase,
-    _input_object_type: Arc<hir::InputObjectTypeDefinition>,
+pub fn check_enum_value(
+    db: &dyn ValidationDatabase,
+    enum_val: hir::EnumValueDefinition,
 ) -> Vec<ApolloDiagnostic> {
-    // TODO: validate extensions
-    // Not checking the `input_values` here as those are checked as fields elsewhere.
-    vec![]
+    db.check_directives(
+        enum_val.directives().to_vec(),
+        hir::DirectiveLocation::EnumValue,
+    )
 }
 
-pub fn check_schema_definition(
+pub fn check_input_object_type_definition(
     db: &dyn ValidationDatabase,
-    schema_def: Arc<hir::SchemaDefinition>,
+    input_obj: Arc<hir::InputObjectTypeDefinition>,
 ) -> Vec<ApolloDiagnostic> {
     let mut diagnostics = Vec::new();
 
-    // TODO: validate extensions
-    for directive in schema_def.directives() {
-        diagnostics.extend(db.check_directive(directive.clone()));
-    }
+    diagnostics.extend(db.check_input_values(
+        (*input_obj).clone().input_fields_definition,
```

Comment:
only need to clone the inner thing
```suggestion
        input_obj.input_fields_definition.clone(),
```